_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,354 @@
+
+> info "Note"
+> Similar to a Controlplane provisioning, a Worker provisioning also takes some time, and you can watch the process using steps shared a bit later. This will also apply when you scale Up/Down workers at a later point in time.
+
+This is what a `MachineDeployment` looks like
+```console
+kubectl describe MachineDeployment ${CLUSTER_NAME} -n metal3
+```
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineDeployment
+metadata:
+  [....]
+  ownerReferences:
+  - apiVersion: cluster.x-k8s.io/v1alpha3
+    kind: Cluster
+    name: eko
+    uid: aec0f73b-a068-4992-840d-6330bf943d22
+  resourceVersion: "66257"
+  selfLink: /apis/cluster.x-k8s.io/v1alpha3/namespaces/metal3/machinedeployments/eko
+  uid: f598da43-0afe-44e4-b793-cd5244c13f4e
+spec:
+  clusterName: eko
+  minReadySeconds: 0
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: eko
+      nodepool: nodepool-0
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: eko
+        nodepool: nodepool-0
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfigTemplate
+          name: eko-workers
+      clusterName: eko
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: Metal3MachineTemplate
+        name: eko-workers
+      version: v1.18.0
+status:
+  observedGeneration: 1
+  phase: ScalingUp
+  replicas: 1
+  selector: cluster.x-k8s.io/cluster-name=eko,nodepool=nodepool-0
+  unavailableReplicas: 1
+  updatedReplicas: 1
+```
+
+To check the status we can follow steps similar to ControlPlane's case :
+```console
+kubectl get bmh -n metal3 -w
+```
+> We can see the live status of the node being provisioned
+
+```console
+kubectl get Machine -n metal3 -w
+```
+> This shows the status of Machines associated with workers, apart from the one for Controlplane, and we can watch the status of provisioning under PHASE 
+
+```console
+sudo virsh net-dhcp-leases baremetal
+```
+> To get the nodes IP
+
+```console
+ssh metal3@{master-node-ip}
+```
+> To check if its added to the cluster
+
+```console
+ssh metal3@{node-ip}
+```   
+> If you want to login to the node
+
+We can add or remove workers to the cluster, we can scale up the MachineDeployment up or down, in this example we are adding 2 more worker nodes
+
+```console
+kubectl scale --replicas=3 MachineDeployment ${CLUSTER_NAME} -n metal3
+```
+
+<br/>
+<br/>
+
+### Deprovisioning
+
+All of the previous components have corresponding deprovisioning scripts which use config files, in the previously mentioned manifest directory, and use them to clean up worker, controlplane and cluster.
```

Comment:
Maybe add an example of deprovisioning?

Reply:
Included an example and added a bit more explanation.